packages/x-components/src/x-modules/history-queries/store/getters/history-queries-with-results.getter.ts

```diff
@@ -0,0 +1,18 @@
+import { HistoryQueriesXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link HistoryQueriesGetters.historyQueriesWithResults} getter.
+ *
+ * @param state - Current {@link https://vuex.vuejs.org/guide/state.html | state} of the history
+ * queries module.
+ *
+ * @param getters - Current {@link https://vuex.vuejs.org/guide/getters.html | getters} of the
+ * history queries module.
+ *
+ * @returns The history queries that have results at the moment they were requested.
+ *
+ * @public
+ */
+// eslint-disable-next-line max-len
+export const historyQueriesWithResults: HistoryQueriesXStoreModule['getters']['historyQueriesWithResults'] =
+  (state, { historyQueries }) => historyQueries.filter(({ totalResults }) => totalResults !== 0);
```

Comment:
Yes but I put it like this intentionally, because meanwhile the request has not succeeded the value will be `undefined` and in that case I think the safest approach is to not remove it from the view. What do you think? Does it make sense or is it better the other way around?

Reply:
I would say that it is safer the other way around.
If there is some error with the request, the history query will be rendered anyway, so it will make the step of not updating the query if the request fails useless.
And even if you add something to handle the request error and hide the history query, It seems a bit odd to remove it just a bit after it was rendered.